src/import/pull-common.c

```diff
@@ -637,6 +652,87 @@ int pull_job_restart_with_sha256sum(PullJob *j, char **ret) {
         return 1;
 }
 
+int signature_style_from_url(const char *url, SignatureStyle *ret, char **fn) {
+        _cleanup_free_ char *last = NULL;
+        SignatureStyle style;
+        int r;
+
+        assert(url);
+        assert(ret);
+        assert(fn);
+
+        /* Determines which kind of signature style is appropriate for this url */
+
+        r = import_url_last_component(url, &last);
+        if (r < 0)
+                return r;
+
+        style = signature_style_from_filename(last);
+        if (style >= 0) {
+                *fn = TAKE_PTR(last);
+                *ret = style;
+                return 0;
+        }
+
+        return style;
+}
+
+int pull_job_restart_with_signature(PullJob *j, char **ret) {
+        _cleanup_free_ char *last = NULL;
+        SignatureStyle style;
+        int r;
+
+        assert(j);
+
+        /* Generic implementation of a PullJobNotFound handler, that restarts the job requesting a different
+         * signature file. After the initial file, additional *.sha256.gpg, SHA256SUMS.gpg and SHA256SUMS.asc
+         * are tried in this order. */
+
+        r = signature_style_from_url(j->url, &style, &last);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine signature style of URL '%s': %m", j->url);
+
+        switch (style) {
+
+        case SIGNATURE_ASC_PER_DIRECTORY: /* Nothing to do anymore */
+                return 0;
+
+        case SIGNATURE_ASC_PER_FILE: { /* Try .sha256.gpg next */
+                char *ext;
+
+                log_debug("Got 404 for %s, now trying to get .sha256.gpg instead.", j->url);
+
+                ext = endswith(last, "asc");
```

Comment:
I'd like to make `".asc"`, that is, please refuse `fooasc` or so, for safety.

Reply:
The check that this is `.sha256.asc` is done 4 lines above, so `fooasc` is impossible.

Reply:
Yes, I know. It is just a paranoia. But still. Please.